### ISSUE TITLE
JSON toast messages when don't have end dates and marking an event complete

### ISSUE THREAD

**Issue by JChipault** [USGS-WiM/whispersservices_django#283]
NWHC epi staff Anne Ballmann was trying to mark 200136 as complete and got the toast message in screen shot below. She then combed through the data to see what she might have been missing and noticed that she didn't have end dates on some locations so she added end dates and then tried switching it to complete and then it worked. Was the toast messages supposed to tell her that she was missing end dates?
![image](https://user-images.githubusercontent.com/49287076/65341928-33603b00-db97-11e9-8285-ed61bb086bac.png)

I (Jenny) then went into 200136, marked it as incomplete, removed the end date from the Pine Lake location, tried to mark it as complete and got a slightly different error message.
Error. Event not updated. Error message: SyntaxError: JSON.parse: unexpected character at line 1 column 2 of the JSON data

I then added the end date back in and tried marking it as complete and it worked fine (no toast error message)

**Comment by BlakeDraper** [comment]
@aaronstephenson  we have an EROS problem again, it looks like to me. This HTML specified URL not found page is telltale server issue. I guess we need to forward them the details of this request and ask that they identify whatever it is about it that that are not allowing, and allow it. 

![image](https://user-images.githubusercontent.com/6978380/65343896-ce5b1400-db9b-11e9-93d6-acd6270c52a4.png)

**Comment by BlakeDraper** [comment]
The JSON request:
```
{
  "id": 200136,
  "event_reference": "",
  "event_type": 1,
  "complete": true,
  "public": true,
  "staff": 3,
  "event_status": 7,
  "quality_check": null,
  "legal_status": 1,
  "legal_number": "",
  "new_read_collaborators": [],
  "new_write_collaborators": []
}

the request URL: https://whispers.usgs.gov/whispersservices/events/200136/
```

**Comment by aaronstephenson** [comment]
Support email sent to EROS.

**Comment by aaronstephenson** [comment]
This was fixed by EROS removing all of their WAF's "active" JSON validation.

**Comment by JChipault** [comment]
@aaronstephenson Just repeated my steps of marking it incomplete, removing an end date, and then trying to mark it complete to see if it gives me a pop-up message that would make sense to a user and direct them to add an end date before marking it complete and it gave me: **Error. Event not updated. Error message: SyntaxError: JSON.parse: unexpected character at line 1 column 2 of the JSON data**

**Comment by aaronstephenson** [comment]
@JChipault was this in production or test?

**Comment by JChipault** [comment]
production.... so maybe this update is coming in a future production release?

**Comment by aaronstephenson** [comment]
No, the back end code is the same in both environments. I just need to know which environment when I talk to EROS, because they have different profiles in the WAF.

**Comment by JChipault** [comment]
thanks

**Comment by aaronstephenson** [comment]
OK, I found the cause of the problem. The error message coming back from the web services is plain text in an array, when it should be formatted as JSON, and so the web client cannot parse the response, hence the syntax error. I will fix this before the end of the day.

**Comment by aaronstephenson** [comment]
This is now fixed in 1.3.8 61aa618
